sdk/storage/storage-queue/src/QueueServiceClient.ts

```diff
@@ -276,4 +226,58 @@ export class QueueServiceClient extends StorageClient {
       ...options
     });
   }
+
+  async *listSegments(
+    marker: string | undefined = undefined,
+    options: ServiceListQueuesSegmentOptions = {}
+  ): AsyncIterableIterator<Models.ServiceListQueuesSegmentResponse> {
+    let listQueuesResponse;
+    do {
+      listQueuesResponse = await this.listQueuesSegment(marker, options);
+      marker = listQueuesResponse.nextMarker;
+      yield await listQueuesResponse;
+    } while (marker);
+  }
+
+  async *listItems(
+    options: ServiceListQueuesSegmentOptions = {}
+  ): AsyncIterableIterator<Models.QueueItem> {
+    let marker: string | undefined;
+    for await (const segment of this.listSegments(marker, options)) {
+      yield* segment.queueItems;
+    }
+  }
+
+  public listQueues(options: ServiceListQueuesSegmentOptions = {}) {
```

Comment:
>ServiceListQueuesSegmentOptions [](start = 29, length = 31)

`ServiceListQueuesSegmentOptions` already supports `pageSize`, seems conflicts with `byPage()` parameters

Reply:
Yeah, I was talking with Brian about it on Monday.
@jeremymeng any good thoughts on this?

Reply:
Added a new interface `ServiceListQueuesOptions` which is basically `ServiceListQueuesSegmentOptions` without `maxResults`.